### PR TITLE
Prac 07 feedback

### PR DESCRIPTION
**Pull request by AxyutitsArtyom**
@dianabukeikhanova
@amritakhatri

### CODE REVIEW

File: prac_07/project_management.py [hunk 1]
```diff
@@ -27,122 +99,126 @@ def main():
...
+    for i in projects:
+        if i.completion_percentage == 100:
+            print("  " + str(i))
 
 
 def get_valid_choice():
+    """Get valid choice from user"""
     choice = input(">>> ").upper()
     while choice not in VALID_CHOICES:
         print("Invalid choice")
```
Comment (DianaBukeikhanova) [inline prac_07/project_management.py:234]: Good code, all errors were catch.

File: prac_07/guitars.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            print(f"{name} ({year}) : ${cost} added.")
+
+    print("These are my guitars:")
+    print_guitars(guitars)
+
+
+def print_guitars(guitars):
+    """Display sorted collection of guitars"""
+    guitars.sort(key=lambda x: x.year)
+    for i, guitar in enumerate(guitars, 1):
```
Comment (DianaBukeikhanova) [inline prac_07/guitars.py:39]: Better to each if we did used and learned lambda.

**Comment by amritakhatri** [comment]
good catch, easy to understand

File: prac_07/project_management.py [hunk 3]
```diff
@@ -45,40 +45,39 @@ def main():
...
-                                 f"{project.priority}	"
-                                 f"{project.cost_estimate}	"
-                                 f"{project.completion_percentage}")
+        with open("projects.txt", "w") as input_file:
+            save_data_to_file(input_file, projects)
+
+
+def save_data_to_file(input_file, projects):
+    input_file.write("Name	Start Date	Priority	Cost Estimate	Completion Percentage\n")
+    for project in projects:
```
Comment (DianaBukeikhanova) [inline prac_07/project_management.py:60]: We use out_file for write or append.

File: prac_07/project_management.py [hunk 4]
```diff
@@ -45,40 +45,39 @@ def main():
...
-            input_file.write("Name	Start Date	Priority	Cost Estimate	Completion Percentage")
-            for project in projects:
-                input_file.write(f"{project.name}	"
-                                 f"{project.start_date}	"
-                                 f"{project.priority}	"
-                                 f"{project.cost_estimate}	"
-                                 f"{project.completion_percentage}")
+        with open("projects.txt", "w") as input_file:
+            save_data_to_file(input_file, projects)
+
```
Comment (DianaBukeikhanova) [inline prac_07/project_management.py:56]: out_file for write and append

File: prac_07/guitar.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __str__(self):
+        return f"{self.name} ({self.year}) : ${self.cost}"
+
+    def __lt__(self, other):
+        return self.year < other
+
+    def get_age(self):
+        return 2024 - self.year
+
```
Comment (DianaBukeikhanova) [inline prac_07/guitar.py:18]: Use constants for 2024

File: prac_07/guitar.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    def is_vintage(self):
+        if self.get_age() >= 50:
+            return True
```
Comment (DianaBukeikhanova) [inline prac_07/guitar.py:21]: also magical number

**Pull request merged**